staging/src/k8s.io/apiserver/pkg/util/proxy/websocket.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/httpstream/wsstream"
+	constants "k8s.io/apimachinery/pkg/util/remotecommand"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/klog/v2"
+)
+
+const defaultIdleConnectionTimeout = 4 * time.Hour
```

Comment:
is this value related to something?

Reply:
I think this is inherited from this config in `staging/src/k8s.io/kubelet/pkg/cri/streaming/server.go`

> 	// How long to leave idle connections open for.
>	StreamIdleTimeout time.Duration

Reply:
Yes. This websocket server code is based on the code in the `kubelet` (and container runtimes), and the `IdleConnectionTimeout` comes from there.

Reply:
but we should explain or give some comments to understand why this magic value

Reply:
I've added some comments for this value. Please let me know what you think.